src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -13,79 +11,93 @@ import { ApprovalState } from 'hooks/useApproveCallback'
-  const parsedValue = new Fraction(value, '100')
-
-  // divide maxValue with parsed value to get invest amount
-  return maxValue.multiply(parsedValue).asFraction
+enum ErrorMsgs {
+  Balance = 'Insufficient balance to cover input investment amount. Adjust the amount or go back to remove this investment option',
+  Input = 'Input amount is bigger than available investment amount',
 }
 
 export default function InvestOption({ approveData, claim, optionIndex }: InvestOptionProps) {
   const { currencyAmount, price, cost: maxCost } = claim
   const { updateInvestAmount } = useClaimDispatchers()
-  const { investFlowData } = useClaimState()
+  const { investFlowData, activeClaimAccount } = useClaimState()
 
   const { handleSetError, handleCloseError, ErrorModal } = useErrorModal()
 
-  const investedAmount = useMemo(() => investFlowData[optionIndex].investedAmount, [investFlowData, optionIndex])
+  const { account } = useActiveWeb3React()
 
-  const [percentage, setPercentage] = useState<string>(INVESTMENT_STEPS[0])
+  const [percentage, setPercentage] = useState<string>('0')
+  const [typedValue, setTypedValue] = useState<string>('0')
+  const [inputError, setInputError] = useState<string>('')
 
-  const { account } = useActiveWeb3React()
+  const investedAmount = investFlowData[optionIndex].investedAmount
 
   const token = currencyAmount?.currency
   const decimals = token?.decimals
   const balance = useCurrencyBalance(account || undefined, token)
 
-  const handleStepChange = useCallback(
+  const isSelfClaiming = account === activeClaimAccount
+
+  // on invest max amount click handler
+  const setMaxAmount = useCallback(() => {
+    if (!maxCost || !balance) {
+      return
+    }
+
+    const value = maxCost.greaterThan(balance) ? balance : maxCost
+    const amount = value.quotient.toString()
+
+    updateInvestAmount({ index: optionIndex, amount })
+    setTypedValue(formatSmart(value, decimals, { smallLimit: undefined }) || '')
+    setPercentage('100')
+  }, [balance, decimals, maxCost, optionIndex, updateInvestAmount])
+
+  // on input field change handler
+  const onInputChange = useCallback(
     (value: string) => {
-      if (!maxCost || !balance) {
+      setTypedValue(value)
+      setInputError('')
+
+      // parse to CurrencyAmount
+      const parsedAmount = tryParseAmount(value, token)
+
+      // no amount/necessary params, return 0
+      if (!parsedAmount || !maxCost || !balance || !token) {
+        updateInvestAmount({ index: optionIndex, amount: '0' })
+        setPercentage('0')
         return
       }
 
-      const scaledCurrencyAmount = _scaleValue(maxCost, value)
-
-      updateInvestAmount({ index: optionIndex, amount: scaledCurrencyAmount.quotient.toString() })
-      setPercentage(value)
-    },
-    [balance, maxCost, optionIndex, updateInvestAmount]
-  )
+      // calculate percent
+      const maxValue = maxCost.greaterThan(balance) ? balance : maxCost
```

Comment:
I think `maxValue` should only be used for the max button action.

For the percentage we should use `maxCost`

If your balance is < maxCost you'll never reach 100% of investment opportunity

Reply:
i dont understand, feel free to commit tho im done here for the night see ya 2morrow

Reply:
Addressed on https://github.com/gnosis/cowswap/pull/2206